.github/workflows/mega-linter.yaml

```diff
@@ -0,0 +1,44 @@
+---
+# MegaLinter GitHub Action configuration file
+# More info at https://megalinter.github.io
+name: MegaLinter
+
+on:
+  pull_request:
+    branches: [main]
+
+concurrency:
+  group: ${{ github.ref }}-${{ github.workflow }}
+  cancel-in-progress: true
+
+jobs:
+  build:
+    name: MegaLinter
+    runs-on: ubuntu-latest
+    steps:
+      # Git Checkout
+      - name: Checkout Code
+        uses: actions/checkout@v3
+
+      # MegaLinter
+      - name: MegaLinter
+        id: ml
+        # You can override MegaLinter flavor used to have faster performances
+        # More info at https://megalinter.github.io/flavors/
+        # uses: oxsecurity/megalinter@v6
+        uses: oxsecurity/megalinter/flavors/terraform@03986e6993ccf699a22451118520680b438e7d2a
+        env:
+          # All available variables are described in documentation
+          # https://megalinter.github.io/configuration/
+          # VALIDATE_ALL_CODEBASE: ${{ github.event_name == 'push' && github.ref == 'refs/heads/main' }} # Validates all source when push on main, else just the git diff with main. Override with true if you always want to lint all sources
```

Comment:
outcommented code is the root to all that is evil  
better just remove it

Reply:
The code is from the pippiio tempalte, but i will clean it up:
https://github.com/pippiio/template/blob/main/.github/workflows/mega-linter.yml